Remove the JSXC/Prosody integration.

skyschub opened the pull request:
As per discussion on https://discourse.diasporafoundation.org/t/removing-diaspora-s-current-chat-integration/2718, nobody raised serious concerns or objections. Given future plans, we do not think having an unfinished implementation of something that likely will not get finished in the current form is worth it. So let's get rid of it.

I deliberately ignored all codestyle warnings. As this PR is exclusively removing things, I felt better not touching anything. Happy to change things if someone feels like it.

When merged, this PR needs documentation in the Changelog that explains the implications.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -61,9 +53,7 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' ),
-      title_chat: Diaspora.I18n.t( 'chat.title' ),
-      chat_enabled: this.chatEnabled()
+      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' )
```
- Identifier 'title_miscellaneous' is not in camel case.
- There should be no spaces inside this paren.
- Strings must use doublequote.

diaspora-code-review commented on spec/integration/exporter_spec.rb:
```diff
@@ -46,12 +46,10 @@
             {
-              "name":             "generic",
-              "chat_enabled":     false
+              "name":             "generic"
```
Layout/AlignHash: Align the elements of a hash literal if they span more than one line.

diaspora-code-review commented on spec/integration/exporter_spec.rb:
```diff
@@ -46,12 +46,10 @@
             {
-              "name":             "Work",
-              "chat_enabled":     false
+              "name":             "Work"
```
Layout/AlignHash: Align the elements of a hash literal if they span more than one line.

diaspora-code-review commented on spec/serializers/export/aspect_serializer_spec.rb:
```diff
@@ -6,8 +6,7 @@
     expect(serializer.attributes).to eq(
-      name:         aspect.name,
-      chat_enabled: aspect.chat_enabled
+      name:         aspect.name
```
Layout/ExtraSpacing: Unnecessary spacing detected.

SuperTux88 reviewed (changes requested):
Overall this looks good to me. I know you ignored codestyle-warnings, but I would still suggest fixing the last 3 warnings about alignment and spacing (I think that improves code quality at these places with only little changes of whitespace), I'm fine with ignoring the first one about camelcase and stuff.

Also, I found the one below, which I think is worth fixing, even not directly related to chat removing.

denschub commented on spec/integration/migration_service_spec.rb:
```diff
@@ -148,8 +148,8 @@
-          {"name":"Friends","chat_enabled":true},
-          {"name":"Friends","chat_enabled":false}
+          {"name":"Friends"},
+          {"name":"Friends"}
```
I... have no idea why that's there. Probably an oversight from adding the chat tests and not noticing the names.

skyschub committed:
Remove the JSXC/Prosody integration.

As per discussion on https://discourse.diasporafoundation.org/t/removing-diaspora-s-current-chat-integration/2718, nobody raised serious concerns or objections. Given future plans, we do not think having an unfinished implementation of something that likely will not get finished in the current form is worth it. So let's get rid of it.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_private_posts: Diaspora.I18n.t( 'private_posts.title' ),
-      title_public_posts: Diaspora.I18n.t( 'public_posts.title' ),
-      title_resharing_posts: Diaspora.I18n.t( 'resharing_posts.title' ),
+      title_header: Diaspora.I18n.t("title_header"),
```
Identifier 'title_header' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_public_posts: Diaspora.I18n.t( 'public_posts.title' ),
-      title_resharing_posts: Diaspora.I18n.t( 'resharing_posts.title' ),
+      title_header: Diaspora.I18n.t("title_header"),
+      title_getting_help: Diaspora.I18n.t("getting_help.title"),
```
Identifier 'title_getting_help' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_resharing_posts: Diaspora.I18n.t( 'resharing_posts.title' ),
+      title_header: Diaspora.I18n.t("title_header"),
+      title_getting_help: Diaspora.I18n.t("getting_help.title"),
+      title_account_and_data_management: Diaspora.I18n.t("account_and_data_management.title"),
```
Identifier 'title_account_and_data_management' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_header: Diaspora.I18n.t("title_header"),
+      title_getting_help: Diaspora.I18n.t("getting_help.title"),
+      title_account_and_data_management: Diaspora.I18n.t("account_and_data_management.title"),
+      title_aspects: Diaspora.I18n.t("aspects.title"),
```
Identifier 'title_aspects' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_getting_help: Diaspora.I18n.t("getting_help.title"),
+      title_account_and_data_management: Diaspora.I18n.t("account_and_data_management.title"),
+      title_aspects: Diaspora.I18n.t("aspects.title"),
+      title_mentions: Diaspora.I18n.t("mentions.title"),
```
Identifier 'title_mentions' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_account_and_data_management: Diaspora.I18n.t("account_and_data_management.title"),
+      title_aspects: Diaspora.I18n.t("aspects.title"),
+      title_mentions: Diaspora.I18n.t("mentions.title"),
+      title_pods: Diaspora.I18n.t("pods.title"),
```
Identifier 'title_pods' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_aspects: Diaspora.I18n.t("aspects.title"),
+      title_mentions: Diaspora.I18n.t("mentions.title"),
+      title_pods: Diaspora.I18n.t("pods.title"),
+      title_posts_and_posting: Diaspora.I18n.t("posts_and_posting.title"),
```
Identifier 'title_posts_and_posting' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_mentions: Diaspora.I18n.t("mentions.title"),
+      title_pods: Diaspora.I18n.t("pods.title"),
+      title_posts_and_posting: Diaspora.I18n.t("posts_and_posting.title"),
+      title_private_posts: Diaspora.I18n.t("private_posts.title"),
```
Identifier 'title_private_posts' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_pods: Diaspora.I18n.t("pods.title"),
+      title_posts_and_posting: Diaspora.I18n.t("posts_and_posting.title"),
+      title_private_posts: Diaspora.I18n.t("private_posts.title"),
+      title_public_posts: Diaspora.I18n.t("public_posts.title"),
```
Identifier 'title_public_posts' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_posts_and_posting: Diaspora.I18n.t("posts_and_posting.title"),
+      title_private_posts: Diaspora.I18n.t("private_posts.title"),
+      title_public_posts: Diaspora.I18n.t("public_posts.title"),
+      title_resharing_posts: Diaspora.I18n.t("resharing_posts.title"),
```
Identifier 'title_resharing_posts' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' ),
-      title_chat: Diaspora.I18n.t( 'chat.title' ),
-      chat_enabled: this.chatEnabled()
+      title_sharing: Diaspora.I18n.t("sharing.title"),
```
Identifier 'title_sharing' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      title_chat: Diaspora.I18n.t( 'chat.title' ),
-      chat_enabled: this.chatEnabled()
+      title_sharing: Diaspora.I18n.t("sharing.title"),
+      title_tags: Diaspora.I18n.t("tags.title"),
```
Identifier 'title_tags' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
-      chat_enabled: this.chatEnabled()
+      title_sharing: Diaspora.I18n.t("sharing.title"),
+      title_tags: Diaspora.I18n.t("tags.title"),
+      title_keyboard_shortcuts: Diaspora.I18n.t("keyboard_shortcuts.title"),
```
Identifier 'title_keyboard_shortcuts' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/views/help_view.js:
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
+      title_sharing: Diaspora.I18n.t("sharing.title"),
+      title_tags: Diaspora.I18n.t("tags.title"),
+      title_keyboard_shortcuts: Diaspora.I18n.t("keyboard_shortcuts.title"),
+      title_miscellaneous: Diaspora.I18n.t("miscellaneous.title")
```
Identifier 'title_miscellaneous' is not in camel case.

SuperTux88 reviewed (approved):
Thanks for the changes, looks good to me :cookie:

SuperTux88 commented:
Merged to `develop`, thanks :cookie:
